### DISCUSSION TITLE
Output Waveform

### DISCUSSION THREAD

**Discussion by rjo** [CrashOverride85/zc95#73] [General]
Is the output waveform bipolar like an ET232/312?  Anyone got a scope image of it?

**Comment by CrashOverride85** [comment]
Yes - output waveform is bipolar, and quite similar to the MK312 at least.

This is it outputting 150hz / 150us into a 500Ω resistor at full power:

![150us 150hz](https://github.com/CrashOverride85/zc95/assets/56609106/e8dc1f07-41c4-45de-bc55-b8e07774690e)

**Comment by rjo** [comment]
Awesome! thanks for the quick response.

So I'm thinking about using the output board with my own controller. How does the DAC input signal map to a positive/negative output?  I'll dig into the code/design but if there is a quick answer to help me grock it I would appreciate it!

**Comment by CrashOverride85** [comment]
Essentially it doesn't, the DAC controls the amplitude in both directions - so the positive & negative output must be the same amplitude (and the DAC isn't fast enough to change in between pulses). 
However the pulse width of the positive & negative output can be varied separately per pulse (right down to 0), which achieve a similar effect.

The general design (as with the 312) is that per channel, there is 1 PFET that controls amplitude (driven by the DAC via an opamp), and 2x NFETs - for the positive and negative pulse.

**Comment by rjo** [comment]
Ahh got it. Thank you!
